be/src/formats/parquet/column_reader.cpp

```diff
@@ -246,18 +249,73 @@ class MapColumnReader : public ColumnReader {
     std::unique_ptr<ColumnReader> _value_reader;
 };
 
+class StructColumnReader : public ColumnReader {
+public:
+    explicit StructColumnReader(const ColumnReaderOptions& opts) {}
+    ~StructColumnReader() override = default;
+
+    Status init(const ParquetField* field, std::vector<std::unique_ptr<ColumnReader>>&& child_readers) {
+        _field = field;
+        _child_readers = std::move(child_readers);
+        return Status::OK();
+    }
+
+    Status prepare_batch(size_t* num_records, ColumnContentType content_type, vectorized::Column* dst) override {
+        vectorized::NullableColumn* nullable_column = nullptr;
+        vectorized::StructColumn* struct_column = nullptr;
+        if (_field->is_nullable) {
+            DCHECK(dst->is_nullable());
+            nullable_column = down_cast<vectorized::NullableColumn*>(dst);
+            DCHECK(nullable_column->mutable_data_column()->is_struct());
+            struct_column = down_cast<vectorized::StructColumn*>(nullable_column->mutable_data_column());
+        } else {
+            DCHECK(dst->is_struct());
+            struct_column = down_cast<vectorized::StructColumn*>(dst);
+        }
+
+        vectorized::Columns fields_column = struct_column->fields_column();
+
+        DCHECK_EQ(fields_column.size(), _child_readers.size());
+
+        for (size_t i = 0; i < fields_column.size(); i++) {
+            vectorized::Column* child_column = fields_column.at(i).get();
+            RETURN_IF_ERROR(_child_readers.at(i)->prepare_batch(num_records, content_type, child_column));
+        }
+
+        if (_field->is_nullable) {
+            DCHECK(dst->is_nullable());
+            DCHECK(nullable_column != nullptr);
+            // Assume all rows are not null in struct level.
+            // Use subfield's NullableColumn instead.
+            vectorized::NullColumn null_column(fields_column.at(0)->size(), 0);
```

Comment:
I don't think you can use the subfield's null column to express the struct columns's null column, we have nullable struct column all row are not null, but may his subfiled1 is null as [1, 0, 0, 0] &[a, b, c, d], and subfiled2 is null as [0, 0, 0, 0] & [e, f, g, h], the result will be [(null), (b,f), (c,g), (d,h)], but it should be[(null, e), (b,f), (c,g), (d,h)]. Am I right?